pkg/networks/validate.go

```diff
@@ -0,0 +1,107 @@
+package networks
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+	"syscall"
+)
+
+func (config *NetworksConfig) Validate() error {
+	// validate all paths.* values
+	paths := reflect.ValueOf(&config.Paths).Elem()
+	for i := 0; i < paths.NumField(); i++ {
+		// extract YAML name from struct tag; strip options like "omitempty"
+		name := paths.Type().Field(i).Tag.Get("yaml")
+		if i := strings.IndexRune(name, ','); i > -1 {
+			name = name[:i]
+		}
+		path := paths.Field(i).Interface().(string)
+		// varPath will be created securely, but any existing parent directories must already be secure
+		if name == "varRun" {
+			path = findBaseDirectory(path)
+		}
+		err := validatePath(path, name == "varRun")
+		if err != nil {
+			// sudoers file does not need to exist; otherwise `limactl sudoers` couldn't bootstrap
+			if name == "sudoers" && errors.Is(err, os.ErrNotExist) {
+				continue
+			}
+			return fmt.Errorf("networks.yaml field `paths.%s` error: %w", name, err)
+		}
+	}
+	// TODO(jandubois): validate network definitions
+	return nil
+}
+
+// findBaseDirectory removes non-existing directories from the end of the path.
+func findBaseDirectory(path string) string {
+	if _, err := os.Lstat(path); errors.Is(err, os.ErrNotExist) {
+		if path != "/" {
+			return findBaseDirectory(filepath.Dir(path))
+		}
+	}
+	return path
+}
+
+const (
+	RootUID   = 0
+	WheelGID  = 0
+	DaemonGID = 1
```

Comment:
These should be moved to `*_darwin.go`

Reply:
> These should be moved to `*_darwin.go`

Just those constants, or the whole code?

The logic is not really darwin specific; it is just that we only call it on darwin.

Even the constants are kind of universal; except that "wheel" should be called "root" on Linux. I'm fine with using `RootGID`. So if anything we should have platform specific constants for the string values (e.g. `RootGroup = "wheel"` on darwin).

Reply:
> Just those constants, or the whole code?

Either is fine.

At least `DaemonGID=1` is specific to Darwin.
On Linux, the daemon GID is often 2.

Reply:
I haven't split the code into platform-specific pieces, but made it look all user and group names / id dynamically: https://github.com/lima-vm/lima/pull/239/commits/c96f16b802ae8cf3566761717ca423cbbc01d6ef